### ISSUE TITLE
Unexpected token u in JSON at position 0 in Application Code Causes Test Fails and stops tests

### ISSUE THREAD

**Issue by joshmccure** [cypress-io/cypress#1250]
- Operating System: Mac OSX
- Cypress Version: Latest
- Browser Version: Chrome 63

### Is this a Feature or Bug?

Bug

### Current behavior:

When an error in my application is triggered in the console 'SyntaxError: Unexpected token u in JSON at position 0' Tests are stopped and failed. This is failure in the application but should have no affect on whether cypress passed or fails.

SyntaxError: Unexpected token u in JSON at position 0
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at https://www.fakeURLinsertedhere.com/:2382:34

### Desired behavior:

This should not fail or stop the test in any way as the test has not failed.

### How to reproduce:
Insert this error in your application, it will fail. This was confirmed by Jonathan Greenemeier @6eDesign in the Cypress Gitter chat.

#### Test code:

```
describe('Smoke', function () {
  it('.should() - assert that <title> is correct', function () {
    cy.visit('https://www.fakeURLinsertedhere.com.au/')
    cy.title().should('include', 'Ttile')

  })
})
```

```js

```

### Additional Info (images, stack traces, etc)

If there is a workaround in the meantime until this bug is fixed this would be great

**Comment by brian-mann** [comment]
Are you referring to uncaught global error handling? When Cypress fails a test for this reason it literally describes (in the error message) that you can turn this behavior off and even provides you a link to our docs explaining how to do it.

It goes here...

https://docs.cypress.io/api/events/catalog-of-events.html#Uncaught-Exceptions

In general we believe in the current behavior - uncaught javascript errors **should** fail tests by default. They are problems with your code that Cypress is finding, and should be corrected.

**Comment by josh-mccure** [comment]
Hi Brian, thanks so much for all your work on Cypress. You're doing amazing things, and I love Cypress.

Just referring back to a bug that I opened and you recently closed cypress-io/cypress#1250

I don't think that Cypresses 'uncaught exception handling' is being triggered by my error 'SyntaxError: Unexpected token u in JSON at position 0'. Cypress didn't provide me a link to this feature and when adding the below code to support/index.js it still doesn't handle or ignore the error.

Cypress.on('uncaught:exception', (err, runnable) => {
// returning false here prevents Cypress from
// failing the test
return false
})

I've also tried including it inside the test and handling the exception on a single basis. Here is part of the stack trace below:

SyntaxError: Unexpected token u in JSON at position 0
SyntaxError: Unexpected token u in JSON at position 0
at JSON.parse (<anonymous>)
at https://www.site.com.au/:2382:34
at Object.error (https://www.site.com.au/CMSScripts/Custom/Shared/visitor-api.js:243:9)
at i (https://www.site.com.au/Libraries/jquery-min.js:2:27449)
at Object.fireWith [as rejectWith] (https://www.site.com.au/Libraries/jquery-min.js:2:28213)
at y (https://www.site.com.au/Libraries/jquery-min.js:4:22746)
at XMLHttpRequest.c (https://www.site.com.au/Libraries/jquery-min.js:4:26925)
From previous event:
at run (https://www.site.com.au/__cypress/runner/cypress_runner.js:61873:15)
at Object.cy.(anonymous function) [as visit] (https://www.site.com.au/__cypress/runner/cypress_runner.js:62093:11)
at Context.runnable.fn (https://www.site.com.au/__cypress/runner/cypress_runner.js:62224:20)
at callFnAsync (https://www.site.com.au/__cypress/runner/cypress_runner.js:32115:8)
at Test.Runnable.run (https://www.site.com.au/__cypress/runner/cypress_runner.js:32067:7)
at https://www.site.com.au/__cypress/runner/cypress_runner.js:65088:28
From previous event:
at Object.onRunnableRun (https://www.site.com.au/__cypress/runner/cypress_runner.js:65087:20)
at $Cypress.action (https://www.site.com.au/__cypress/runner/cypress_runner.js:60578:51)
at Test.Runnable.run (https://www.site.com.au/__cypress/runner/cypress_runner.js:64238:20)
at Runner.runTest (https://www.site.com.au/__cypress/runner/cypress_runner.js:32555:10)
at https://www.site.com.au/__cypress/runner/cypress_runner.js:32661:12
at next (https://www.site.com.au/__cypress/runner/cypress_runner.js:32475:14)
at https://www.site.com.au/__cypress/runner/cypress_runner.js:32485:7
at next (https://www.site.com.au/__cypress/runner/cypress_runner.js:32417:14)
at https://www.site.com.au/__cypress/runner/cypress_runner.js:32453:5
at timeslice (https://www.site.com.au/__cypress/runner/cypress_runner.js:27694:27)
_

**Comment by brian-mann** [comment]
Can you provide test code and a screenshot of what is happening
